### PR TITLE
[Feature] Add Cancel Request Modals for New Replacement and Renewal Requests

### PR DESCRIPTION
**Pull request by jennifertsai**
## Notion ticket link

[Add Cancel Request Modals for New Replacement and Renewal Requests](https://www.notion.so/uwblueprintexecs/Add-Cancel-Request-Modals-for-New-Replacement-and-Renewal-Requests-aa26111a9ffe45e4b3832bc8e7ec76ab)

## Implementation description
* Created modal for when users click "cancel" on the bottom of create renewal and create replacement forms
* Modal closes when user selects "close" 
* Modal redirects user to /admin when user selects "cancel"

Animation

image

## Notes
* 

## Checklist
- [x] My PR name is descriptive, is in imperative tense and starts with one of the following: `[Feature]`,`[Improvement]` or `[Fix]`,
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the RCD team on GitHub, or specific people who are associated with this ticket

### CODE REVIEW

**Review by OustanDing — COMMENTED** [review]
Great work! There's a small change to the text and buttons in the modal - could you please make those small changes based on the Figma?

File: components/admin/requests/modals/CancelCreateRequestModal.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+} from '@chakra-ui/react'; // Chakra UI
+import Link from 'next/link'; // Link
+import { ReactNode } from 'react'; // React JSX Type
+
+type CancelRequestModalProps = {
+  readonly type: 'replacement' | 'renewal';
+  readonly children: ReactNode;
+};
+
+export default function CancelCreateRequestModal({ type, children }: CancelRequestModalProps) {
```
Comment (EmilioMena) [inline components/admin/requests/modals/CancelCreateRequestModal.tsx:21]: Could you add a small jsdoc comment for this component, would go a long way for reusability!

File: pages/admin/request/create-replacement.tsx [hunk 2]
```diff
@@ -172,21 +173,22 @@ export default function CreateReplacement() {
...
+              <Stack direction="row" justifyContent="space-between">
+                <CancelCreateRequestModal type="replacement">
+                  <Button
+                    bg="secondary.critical"
+                    _hover={{ bg: 'secondary.criticalHover' }}
+                    marginRight="20px"
+                    height="48px"
+                    width="149px"
+                  >
+                    <Text textStyle="button-semibold">Discard</Text>
```
Comment (EmilioMena) [inline pages/admin/request/create-replacement.tsx:185]:
```suggestion
                    <Text textStyle="button-semibold">Discard request</Text>
```

File: pages/admin/request/create-renewal.tsx [hunk 3]
```diff
@@ -194,23 +195,24 @@ export default function CreateRenewal() {
...
+              <Stack direction="row" justifyContent="space-between">
+                <CancelCreateRequestModal type="renewal">
+                  <Button
+                    bg="secondary.critical"
+                    _hover={{ bg: 'secondary.criticalHover' }}
+                    marginRight="20px"
+                    height="48px"
+                    width="149px"
+                  >
+                    <Text textStyle="button-semibold">Discard</Text>
```
Comment (EmilioMena) [inline pages/admin/request/create-renewal.tsx:207]:
```suggestion
                    <Text textStyle="button-semibold">Discard request</Text>
```

**Review by EmilioMena — COMMENTED** [review]
Great work Jennifer! Just left a comment about adding a jsdoc and a couple of small discrepancies I noticed from figma. 
Also, does this need a design review @OustanDing ?

**Comment by OustanDing** [comment]
@EmilioMena No design review needed!

**Review by EmilioMena — COMMENTED** [review]
Thanks for making those changes! I see there's a few merge conflicts pending but other than that this looks good to me 🚀

**Review by OustanDing — APPROVED** [review]
LGTM!

**Review by EmilioMena — APPROVED** [review]

**Pull request merged**